### PR TITLE
rptun: Strip rpmsg and rptun. Extract common rpmsg api in rptun.c to rpmsg.c, rpmsg provide rpmsg_ops to each backend for specific implementation.

### PR DESCRIPTION
**Pull request by wyr8899**
## Summary
nuttx/driver/rpmsg: new folder, extract common rpmsg api in rptun.c to rpmsg.c. 
rpmsg provide rpmsg_ops to each backend for specific implementation.
## Impact
Clarify the functions of rptun and rpmsg so that rpmsg can adapt to other types of transport layers.
## Testing
Tested in sim vela, 86panel, n62. Verification function is normal.

### CODE REVIEW

File: include/nuttx/rpmsg/rpmsg.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  FAR const struct rpmsg_ops_s *ops;
+  struct rpmsg_device          rdev[0];
+};
+
+typedef CODE int (*rpmsg_foreach_t)(FAR struct rpmsg_s *rpmsg,
+                                    FAR void *arg);
+
+/**
+ * struct rpmsg_ops_s - Rpmsg device operations
+ * @get_cpuname: get cpu name.
```
Comment (raiden00pl) [inline include/nuttx/rpmsg/rpmsg.h:52]: please remove `@` from comments. See https://github.com/apache/nuttx/pull/11363

File: include/nuttx/rptun/rptun.h [hunk 2]
```diff
@@ -30,7 +30,9 @@
 #ifdef CONFIG_RPTUN
 
 #include <nuttx/fs/ioctl.h>
-#include <openamp/open_amp.h>
+#include <nuttx/rptun/rptun_ping.h>
```
Comment (raiden00pl) [inline include/nuttx/rptun/rptun.h:33]: there is no such file in upstream

**Review by raiden00pl — COMMENTED** [review]
Good idea to separate rptun and rpmsg

File: include/nuttx/rptun/rptun.h (hunk 2 above)
Comment (raiden00pl) [reply include/nuttx/rptun/rptun.h:33]: I see that this file is added in https://github.com/apache/nuttx/pull/11537 but this line should not be in this PR

Comment (wyr8899) [reply include/nuttx/rptun/rptun.h:33]: Yes, this PR should be submited after #11537.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by CV-Bowen** [comment]
@wyr8899 Let's fix the CI problem.

**Comment by xiaoxiang781216** [comment]
please rebase to the last master the ci break is fixed, @wyr8899

**Pull request merged**